packages/wrapped-keys/src/lib/api/batch-generate-private-keys.ts

```diff
@@ -0,0 +1,73 @@
+import { getKeyTypeFromNetwork } from './utils';
+import { batchGenerateKeysWithLitAction } from '../lit-actions-client';
+import { getLitActionCodeOrCidCommon } from '../lit-actions-client/utils';
+import { storePrivateKey } from '../service-client';
+import {
+  BatchGeneratePrivateKeysActionResult,
+  BatchGeneratePrivateKeysParams,
+  BatchGeneratePrivateKeysResult,
+} from '../types';
+import {
+  getFirstSessionSig,
+  getPkpAccessControlCondition,
+  getPkpAddressFromSessionSig,
+} from '../utils';
+
+/**
+ *  TODO: Document batch behaviour
+ * @param { BatchGeneratePrivateKeysParams } params Parameters to use for generating keys and optionally signing messages
+ *
+ * @returns { Promise<BatchGeneratePrivateKeysResult> } - The generated keys and, optionally, signed messages
+ */
+export async function batchGeneratePrivateKeys(
+  params: BatchGeneratePrivateKeysParams
+): Promise<BatchGeneratePrivateKeysResult> {
+  const { pkpSessionSigs, litNodeClient } = params;
+
+  const sessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(sessionSig);
+
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  const { litActionCode, litActionIpfsCid } = getLitActionCodeOrCidCommon(
```

Comment:
Actually, this is named accurately -- one of these two values will _always_ be falsy -- if the code exists, it will be returned and `litActionIpfsCid` will be undefined, and if the code doesn't exist then `litActionIpfsCid` will be set but the code will be falsy.